drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/PersistableRunner.java

```diff
@@ -578,11 +578,18 @@ public RequestContext execute( Executable executable, RequestContext context ) {
 
             // Open the entity manager before the transaction begins.
             PersistenceContext persistenceContext = jpm.getApplicationScopedPersistenceContext();
-
+            // We flag the current persistence runner
+            final String DROOLS_PARENT_RUNNER = "DROOLS_PARENT_RUNNER";
+            boolean isParentRunner = txm.getResource(DROOLS_PARENT_RUNNER) == null;
             boolean transactionOwner = false;
+
             try {
+                if (isParentRunner) {
+                    txm.putResource(DROOLS_PARENT_RUNNER, new Object());
```

Comment:
Just a silly suggestion. Maybe this is a little clearer

```suggestion
                    txm.putResource(DROOLS_PARENT_RUNNER, Boolean.TRUE);
```